### PR TITLE
Fix HTTP links in the site.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
The site was broken with the active mixed content.

### CODE REVIEW

**Comment by kkm000** [comment]
Thanks. I see what you mean, using https the site looks broken. But actually the site can be accessed using either http or https. Will this work if someone opens it with http?

**Comment by kkm000** [comment]
I think that a better, and actually the most correct way to handle this is to omit the protocol spec entirely: https://stackoverflow.com/q/4831741/1149924.

File: docs/tools/templates/template.cshtml [hunk 1]
```diff
@@ -7,23 +7,23 @@
...
-      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
+      <script src="https://html5shim.googlecode.com/svn/trunk/html5.js"></script>
     <![endif]-->
   </head>
   <body>
     <div class="container">
       <div class="masthead">
         <ul class="nav nav-pills pull-right">
-          <li><a href="http://fsharp.org">fsharp.org</a></li>
+          <li><a href="https://fsharp.org">fsharp.org</a></li>
```
Comment (kkm000) [inline docs/tools/templates/template.cshtml]: This one is actually good (the site requests https anyway), but the elements of the page should better be protocol-relative.

**Comment by teo-tsirpanis** [comment]
@kkm000, it doesn't matter actually.

__HTTP is legacy.__

[F#'s site](https://fsharp.org/) _cannot_ be accessed from HTTP. GitHub Pages written after June 15, 2016 cannot be accessed from HTTP, and I recommend you to do the same from the Settings tab.

**Comment by kkm000** [comment]
Thank you for the recommendation. Last time I checked, HTTP was still an IETF standard. I do not see a point of arguing here: If a choice is between retaining the compatibility and breaking it, I definitely go with the former.

**Comment by teo-tsirpanis** [comment]
> Last time I checked, HTTP was still an IETF standard.

Well, I admit my use of the word "legacy" was excessive. I had meant that plain HTTP has security issues even for static sites, and is not used in new applications.

Besides, F#'s site has HTTPS enforced. I see no reason not to do the same thing here.

As for compatibility, [TLS 1.2 is now supported by all major browsers](https://community.qualys.com/thread/12421). I find it a little pointless to argue over this. 😛

**Comment by kkm000** [comment]
I'm glad we agree that this point is not worth contention. Naturally, neither me nor you would want to break the site for http access before http is disabled, and this is exactly what this proposed change is doing. So, in the end, since we are speaking of fixing one thing without breaking the other, would you want to make the content links protocol-relative please? If you have philosophical objections to doing that, feel free to drop this change on the floor--no hard feelings at all. In any case, thanks for pointing out that the problem does exist!

**Comment by teo-tsirpanis** [comment]
Actually, I just noticed that __I cannot access the site using HTTP__, which means that _HTTPS is already enforced_.

**Comment by teo-tsirpanis** [comment]
[Omitting the protocol from a URL is considered an anti-pattern](https://stackoverflow.com/a/27999789/4964734) in a time when HTTPS is the norm.

Moreover, these protocol-less URLs do not play well when the site is from a `file:` URL prefix; something quite important, given the site's static nature.

Meanwhile, the site is broken in all modern browsers and _gives the impression that the project is dead_.

**Comment by kkm000** [comment]
First of all, to clarify where we stand, I do not have time or interest or qualification to discuss what is considered "the right" pattern or practice. There are enough of these around, and every one of them is claiming to be "the best" or "the only correct." For one example, the W3C director, someone Timothy Berners-Lee, of whom you have probably heard, exposes multiple pitfalls of the idea (google "Berners-Lee on https", and you'll find the series of his writings, if interested). I have a hunch that, since he has been in fact working with the principles and implementation of the World Wide Web longer than anyone else, his points must not be taken lightly. But, I am repeating myself, I'm asking to please put this outside of scope of this conversation.

My task here is only to make sure that the Web site is accessible to people regardless of the schema they use, http:// or https://. I hope that it would be entirely just to assume that your technical knowledge of this matter equals or surpasses your enthusiasm, and thus should be great, and I need your expertise. I am wondering what would be the impact on the site accessing page parts, such as the scripts, by using the https: schema unconditionally, as you propose, if the case the page itself is accessed over http:. I know that the reverse ("unsecure" parts in a "secure" page) is a really bad scenario, as you reported, which is frowned upon by by IE, and silently prohibited by Chrome and FF, and likely Edge as well. But will all browsers that people realistically use allow "secure" parts in an "unsecure" page without a glitch or a user-visible warning or other possible deterioration in user experience? If that's the case, I'll go with your solution. I think you have a point: as more content providers are turning off http, and some of these scripts might suddenly become unavailable over http from their respective CDNs. *And that's my real, and the only technical concern that I must focus on.*

Could you please locate and provide link(s) to reliable references that would positively confirm that using https for scripts inside an http-opened page is A-OK in practice with major browsers? Is this situation standardized by IETF or W3C? This will tip the balance toward explicit https: schema, which you propose, considering the concern that I cited above. This will be very helpful, and I would really appreciate this, thanks!

**Comment by teo-tsirpanis** [comment]
> Could you please locate and provide link(s) to reliable references that would positively confirm that using https for scripts inside an http-opened page is A-OK in practice with major browsers?

[Here it comes](https://stackoverflow.com/questions/20516584/is-loading-scripts-or-other-resources-via-https-on-a-http-page-problematic). It says that loading HTTPS resources on an HTTP page is not problematic, "\[a\]t least, not on any browsers that remain in popular use."

I also [analyzed the JQuery's CDN TLS versions](https://www.ssllabs.com/ssltest/analyze.html?d=code.jquery.com). The minimum supported version is TLS 1.0, [which is supported in _all_ browsers apart from IE versions lower than 4](https://help.salesforce.com/articleView?id=000220586&language=en_US&type=1).

I also observed behavior in of other fsprojects' sites. Try loading Paket's site in HTTP in both Chrome and Edge. For some reason, Chrome _internally redirects_ to HTTPS, but on Edge, it loads the HTTP site, __with HTTPS resources__ like a charm. Notice that Paket's site has explicit HTTPS resources.

Chrome's behavior when loading the HTTP site

Edge's behavior when loading the HTTP site

Paket's site uses explicit HTTPS resources

I did not find any formal standard regarding HTTPS resources on HTTP pages, but the Stack Overflow answer I provided at the top, the tests on Edge (I did not have other browsers ready) say that it is perfectly OK. Besides, it would make HTTP pages more secure. Why would there be any problems with it?

File: docs/tools/templates/template.cshtml [hunk 2]
```diff
@@ -7,23 +7,23 @@
...
+    <script src="https://code.jquery.com/ui/1.8.23/jquery-ui.js"></script>
+    <script src="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.2.1/js/bootstrap.min.js"></script>
+    <link href="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.2.1/css/bootstrap-combined.min.css" rel="stylesheet">
 
     <link type="text/css" rel="stylesheet" href="@Root/content/style.css" />
     <script type="text/javascript" src="@Root/content/tips.js"></script>
     <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
     <!--[if lt IE 9]>
-      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
+      <script src="https://html5shim.googlecode.com/svn/trunk/html5.js"></script>
```
Comment (kkm000) [inline docs/tools/templates/template.cshtml]: One last thing if you do not mind. While we are at it, please kill the whole HTML5 shim kluge. The URL is dead anyway, as is IE6-8.

**Comment by kkm000** [comment]
Thank you for the research, this is entirely convincing! This is ready to merge, pending one last comment I just added while verifying that the links are in fact alive. Looks like the shim CDN went under together with Google Code, and reviving it makess no sense as IE8 has been no longer supported since early 2016.

> I did not find any formal standard regarding HTTPS resources on HTTP pages

I do not believe this behavior has been standardized really.

**Comment by teo-tsirpanis** [comment]
@kkm000, it is done.

**Comment by kkm000** [comment]
LGTM, thank you!

**Comment by teo-tsirpanis** [comment]
@kkm000, you should also rebuild the site for the changes to have effect.

**Comment by kkm000** [comment]
I know, I know! :)

@sergey-tihon, it does not look like I have write access to the site--can you please help me with that?

**Comment by sergey-tihon** [comment]
@kkm000 you have write access to all branches (`site` = `gh-pages` branch)
It should be as easy as call `.\build.cmd ReleaseDocs` from the console.

I cannot publish from `master` because it does not compile on my win machine
```
       "X:\FsLexYacc\src\FsLex\FsLex.fsproj" (Rebuild target) (1) ->
       (CoreCompile target) ->
         X:\FsLexYacc\src\FsLex\fslexast.fs(129,7): error FS0410: The type 'Position' is less accessible than the value, member or type 'TopCode
       ' it is used in. [X:\FsLexYacc\src\FsLex\FsLex.fsproj]
         X:\FsLexYacc\src\FsLex\fslexast.fs(132,7): error FS0410: The type 'Position' is less accessible than the value, member or type 'BottomC
       ode' it is used in. [X:\FsLexYacc\src\FsLex\FsLex.fsproj]
```

**Pull request merged**